sql/driver/driver_test.go

```diff
@@ -203,6 +206,63 @@ func TestPlaceholders(t *testing.T) {
 	}
 }
 
+func TestinConnectionSettings(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s := server.StartTestServer(nil)
+	url := "https://root@" + s.ServingAddr() + "?certs=test_certs"
+	// Create a new Database called t.
+	db, err := sql.Open("cockroach", url)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := db.Exec(`CREATE DATABASE t`); err != nil {
+		t.Fatal(err)
+	}
+	if err := db.Close(); err != nil {
+		t.Fatal(err)
+	}
+	// Open a new db client against the server with
+	// a fresh set of connections with the new settings.
+	// We create a database 't' above because you cannot
+	// use sql: "SET DATABASE = t" without the database
+	// existing.
+	url += "&database=t"
+	if db, err = sql.Open("cockroach", url); err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup(s, db)
+	var wq sync.WaitGroup
+	numRoutines := 5
+	wq.Add(numRoutines)
+	for i := 0; i < numRoutines; i++ {
+		go func(i int) {
+			defer wq.Done()
+			for r := retry.Start(retry.Options{InitialBackoff: 10 * time.Millisecond, MaxBackoff: 20 * time.Millisecond}); r.Next(); {
+				tx, err := db.Begin()
+				if err != nil {
+					t.Fatal(err)
+					return
+				}
+				// Settings work!
+				if _, err := tx.Exec(fmt.Sprintf(`CREATE TABLE kv%d (k CHAR PRIMARY KEY, v CHAR)`, i)); err != nil {
```

Comment:
a few things:

- when you use a txn, there is no need to use goroutines, since the go sql driver will pin that transaction to a particular connection.
- why bother creating the table multiple times? create it once outside the loop and then `select * from kv` <- unqualified name in a series of "concurrent" transactions. "concurrent" just means the transactions have overlapping lifetimes